File: 2022.01.09/입출력과 사칙연산/python/2588.py
# 배열 뒤집어서 입력받기 ( 리스트로 )
# b부터 반복하면서 b의 LSB -> MSB
# digit : 곱 계산의 자릿수, place_value = (곱 계산 결과의) 합 계산의 자릿수

a = list(map(int,input()))[::-1]
b = list(map(int,input()))[::-1]

place_value = 1
result = 0

for y in b:
    digit = 1
    tmp = 0
    for x in a:
        tmp += x*y*digit
        digit *= 10
    print(tmp)
    result += tmp*place_value
    place_value *= 10
print(result)

File: Boj.단계별/1_입출력과 사칙연산/javascript/2588.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');
const a = input[0];
const b = input[1];
let result = 0;
let place_value = 1;

for (let i = b.length-1 ; i>=0 ; i--){
    let temp = 0;
    let digit = 1;
    for (let j = a.length-1 ; j>=0 ; j--){
        temp += Number(a[j])*Number(b[i])*digit;
        digit *= 10;
    }
    console.log(temp);
    result += temp*place_value;
    place_value *= 10;
}
console.log(result);